### PR TITLE
Removes snapcraft from main repo

### PR DESCRIPTION
**Pull request by tomponline**
It has now moved to https://github.com/canonical/microcloud-pkg-snap/

Also moves unit tests into code tests action.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -62,7 +45,7 @@ jobs:
       SNAPCRAFT_BUILD_ENVIRONMENT: "lxd"
```
[the remark this answers is not in the document]
Comment (tomponline) [reply .github/workflows/tests.yml:45]: I'd be open to that.

[the remark this answers is not in the document]
Comment (tomponline) [reply .github/workflows/tests.yml]: done

**Review by masnax — APPROVED** [review]
Lgtm, thanks!

**Comment by tomponline** [comment]
@masnax could you file PR(s?) to fix the use of these removed functions please?

**Comment by tomponline** [comment]
@masnax also this suggests the go deps are not pinned or are being updated as part of the test (not good for repeatable tests) so can you pin the deps (or not update them in the tests) please. In LXD we periodically update the deps manually in a PR.

**Pull request merged**